src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -258,20 +258,33 @@ public function canFilterAnyColumns($gridField)
             return false;
         }
         $modelClass = $gridField->getModelClass();
-        // note: searchableFields() will return summary_fields if there are no searchable_fields on the model
-        $searchableFields = array_keys($modelClass::singleton()->searchableFields());
-        $summaryFields = array_keys($modelClass::singleton()->summaryFields());
-        sort($searchableFields);
-        sort($summaryFields);
-        // searchable_fields has been explictily defined i.e. searchableFields() is not falling back to summary_fields
-        if ($searchableFields !== $summaryFields) {
-            return true;
-        }
-        // we have fallen back to summary_fields, check they are filterable
-        foreach ($searchableFields as $searchableField) {
-            if ($list->canFilterBy($searchableField)) {
+        if (method_exists($modelClass, 'summaryFields') && method_exists($modelClass, 'searchableFields')) {
+            // note: searchableFields() will return summary_fields if there are no searchable_fields on the model
+            $searchableFields = array_keys($modelClass::singleton()->searchableFields());
+            $summaryFields = array_keys($modelClass::singleton()->summaryFields());
```

Comment:
Should we use `hasMethod` on the singleton? Not sure if it has any benefits here, but every time I see `method_exists` check it comes to mind. Possibly out of scope here, just a suggestion.

```suggestion
        $modelSingleton = $modelClass::singleton();
        if ($modelSingleton->hasMethod('summaryFields') && $modelSingleton->hasMethod('searchableFields')) {
            // note: searchableFields() will return summary_fields if there are no searchable_fields on the model
            $searchableFields = array_keys($modelSingleton->searchableFields());
            $summaryFields = array_keys($modelSingleton->summaryFields());
```

Reply:
Yeah, good point. I had thought of that before but then I though "but then we're assuming the class has `singleton()` and `hasMethod()` on it.
I forgot we have a `singleton` global function, which is already being used to get singletons of this class anyway.